test/classes/initializers/phase1/dynamicThis/genericParent.chpl

```diff
@@ -0,0 +1,49 @@
+
+// This test exists to ensure that the compiler doesn't screw up when fiddling
+// with the type and class ID of 'this' in phase one.
+
+class Parent {
+  type eltType;
+  var x : eltType;
+
+  proc init(type t) {
+    this.eltType = t;
+  }
+
+  proc inheritedMethod() {
+    writeln(this.type:string, ".inheritedMethod()");
+  }
+}
+
+proc foobar(p : Parent(int)) {
+  writeln("foobar(Parent(int))");
+}
+proc foobar(p : Parent) {
+  writeln("foobar(Parent(other))");
+}
+
+class Child : Parent {
+  var y : real;
+
+  proc init(type t) {
+    super.init(t);
+    inheritedMethod();
+    foobar(this);
+    initDone();
+    inheritedMethod();
+  }
+
+  proc inheritedMethod() {
+    writeln(this.type:string, ".inheritedMethod()");
```

Comment:
One quick comment on this test:  I may be missing something, but I think that the definitions of inheritedMethod() on both the parent and the child class are identical which makes me think that we can't tell which was called if something was going wrong (?).

Reply:
Those methods should print the type of 'this' when called, so I think we're fine. Unless you're seeing something that I'm not?

Reply:
Right, but if, during phase 2, we called the `parent.inheritedMethod()` version, wouldn't it print the child's type even if the child version was supposed to have been called?  My thought was that if we put a `(parent)` or `(child)` label into each of the two routines, we could validate not only that the dynamic type was as expected, but also that the correct method was being called.  But again, I may be missing something else.